daemon/vpn/nordlynx/libtelio/libtelio.go

```diff
@@ -288,6 +302,9 @@ func (l *Libtelio) connect(
 	// Check if the connection actually happened. Disconnect if no actual connection was
 	// created within the timeout or until it was canceled.
 	select {
+	case <-connectCtx.Done():
```

Comment:
Maybe in this case it would be possible to use connectCtx as a parent of ctx? Then lines 305-307 could be removed.